v2/inc/c_logging/log_context.h

```diff
@@ -0,0 +1,193 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#ifndef LOG_CONTEXT_H
+#define LOG_CONTEXT_H
+
+#ifdef __cplusplus
+#include <cstdint>
+#include <cstdio>
+#include <cstdlib>
+#else
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#endif
+
+#include "macro_utils/macro_utils.h"
+
+#include "c_logging/log_context_property_value_pair.h"
+#include "c_logging/log_context_property_basic_types.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// This code is intentionally a spike only to showcase an implementation
+// Code with tests shall be written after RFC comments are received
+
+typedef struct LOG_CONTEXT_TAG
+{
+    char* values_string;
+    uint32_t values_string_length;
+    uint8_t* values_data;
+    uint32_t values_data_length;
+    LOG_CONTEXT_PROPERTY_VALUE_PAIR* property_value_pairs_ptr;
+    uint32_t property_value_pair_count;
+    LOG_CONTEXT_PROPERTY_VALUE_PAIR property_value_pairs[];
```

Comment:
>property_value_pairs [](http://example.com/codeflow?start=36&length=20)

this and line 35 are different, or ... ?

Reply:
Yes, one is a variable size array to allocate everything in one shot for a context and the other in ln 35 is an alias.
In case of stack allocated contexts, the property_value_pairs_ptr alias actually points to a stack variable where the pairs are.